pkg/config/config.go

```diff
@@ -311,10 +323,9 @@ func Load(fs *flag.FlagSet, args []string) (*Config, error) {
 		if features.Enabled(fs, featRemoteConfigs) {
 			return LoadRemote(url, expand, c)
 		}
-		if features.Enabled(fs, featDynamicConfig) {
-			if !features.Enabled(fs, featIntegrationsNext) {
-				panic("integrations-next must be enabled for dynamic configuration to work")
-			}
+		if features.Enabled(fs, featDynamicConfig) && !features.Enabled(fs, featIntegrationsNext) {
+			panic("integrations-next must be enabled for dynamic configuration to work")
```

Comment:
nit: I'd prefer only showing a panic to a user if it's from a logic issue that should never happen (ie not the user's fault). For a misconfiguration, I'd like for us to return errors instead.

Reply:
👍